opacities/gray_opacity.hpp

```diff
@@ -0,0 +1,90 @@
+// ======================================================================
+// © 2021. Triad National Security, LLC. All rights reserved.  This
+// program was produced under U.S. Government contract
+// 89233218CNA000001 for Los Alamos National Laboratory (LANL), which
+// is operated by Triad National Security, LLC for the U.S.
+// Department of Energy/National Nuclear Security Administration. All
+// rights in the program are reserved by Triad National Security, LLC,
+// and the U.S. Department of Energy/National Nuclear Security
+// Administration. The Government is granted for itself and others
+// acting on its behalf a nonexclusive, paid-up, irrevocable worldwide
+// license in this material to reproduce, prepare derivative works,
+// distribute copies to the public, perform publicly and display
+// publicly, and to permit others to do so.
+// ======================================================================
+
+#ifndef OPACITIES_GRAY_EMISSIVITY
+#define OPACITIES_GRAY_EMISSIVITY
+
+#include <cassert>
+#include <cmath>
+#include <cstdio>
+
+#include <opac-utils/opac_error.hpp>
+#include <opac-utils/physical_constants.hpp>
+#include <ports-of-call/portability.hpp>
+
+#include "thermal_distributions.hpp"
+
+namespace singularity {
+
+template <typename ThermalDistribution, int NSPECIES> class GrayOpacity {
+public:
+  GrayOpacity(const Real kappa) : kappa_(kappa) {}
```

Comment:
kappa is often an opacity (dimensions cm^2 g^-1) whereas here kappa is an absorption coefficient (dimensions cm^-1). My fault for having done this in earlier code I think. If we're passing in opacities should not `EmissivityPerNuOmega` et al. then return `rho*kappa_*Bnu`?

You could add a separate class, `GrayAbsortionCoefficient`, for the most extreme approximators who want constant optical depth over some distance regardless of material density

Reply:
On second thought implementing `GrayAbsorptionCoefficient` would probably be a waste of time.

Reply:
>  If we're passing in opacities should not EmissivityPerNuOmega et al. then return `rho*kappa_*Bnu`?

Good catch. This is a bug. Thanks. I don't think we need a `GrayAbsorptionCoefficient` though.